src/main/resources/static/js/MainWindow.js

```diff
@@ -1,670 +1,756 @@
+		data : mydata,
+		success : function(response) {
+			var json = JSON.stringify(response, null, 4);
+			var jsonparsed = JSON.parse(json);
+
+			var list = $("#things");
+			list.empty().append(new Option("", "", null, null));
+			for (var i = 0; i < jsonparsed.length; i++) {
+				var option = new Option(jsonparsed[i].name + " ("
+						+ jsonparsed[i].frostId + ")", jsonparsed[i].name
+						+ " (" + jsonparsed[i].frostId + ")", null, null);
+				option.setAttribute("data-value", JSON.stringify(jsonparsed[i],
+						null, 4));
+				list.append(option);
+			}
+			list.trigger("change");
+			addToLog("Things loaded.");
+
+			if (!(fnSuccess == null)) {
+				fnSuccess();
+			}
+		},
+		error : function(e) {
+			addToLog(e.responseText);
+
+		}
+	});
 
+}
 
 var streamData = [];
 
 /**
  * gets all datastreams from the thing with id id
- * @param id id of a thing
+ * 
+ * @param id
+ *            id of a thing
  */
 function getThingStreams(id, cfg, streams) {
 	var url = document.getElementById("serverurlbox").innerText;
-    $.ajax({
-        type: "GET",
-        url: "datastream/all",
-        data: {
-            thingId: id,
-            url: url
-        },
-        success: function (response) {
-            var json = JSON.stringify(response, null, 4);
-            var jsonparsed = JSON.parse(json);
-
-            streamData = [];
-            var stream = {};
-            stream["id"] = "";
-            stream["text"] = "";
-            streamData.push(stream);
-
-            for (var i = 0; i < jsonparsed.length; i++) {
-                stream = {};
-                stream["id"] = jsonparsed[i].name + " (" + jsonparsed[i].frostId + ")";
-                stream["text"] = jsonparsed[i].name + " (" + jsonparsed[i].frostId + ")";
-                stream["data-value"] = JSON.stringify(jsonparsed[i], null, 4);
-                streamData.push(stream);
-            }
-
-            $("#datastreams").empty();
-            if (cfg) {
-                fillStreams(streams);
-            } else {
-                addDatastream();
-            }
-
-        },
-        error: function (e) {
-            addToLog(e.responseText);
-        }
-    });
+	$.ajax({
+		type : "GET",
+		url : "datastream/all",
+		data : {
+			thingId : id,
+			url : url
+		},
+		success : function(response) {
+			var json = JSON.stringify(response, null, 4);
+			var jsonparsed = JSON.parse(json);
+
+			streamData = [];
+			var stream = {};
+			stream["id"] = "";
+			stream["text"] = "";
+			streamData.push(stream);
+
+			for (var i = 0; i < jsonparsed.length; i++) {
+				stream = {};
+				stream["id"] = jsonparsed[i].name + " ("
+						+ jsonparsed[i].frostId + ")";
+				stream["text"] = jsonparsed[i].name + " ("
+						+ jsonparsed[i].frostId + ")";
+				stream["data-value"] = JSON.stringify(jsonparsed[i], null, 4);
+				streamData.push(stream);
+			}
+
+			$("#datastreams").empty();
+			if (cfg) {
+				fillStreams(streams);
+			} else {
+				addDatastream();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['addDatastream' was used before it was defined.](https://app.codacy.com/gh/FraunhoferIOSB/ChillImport/pullRequest?prid=4781409)